pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -101,9 +116,16 @@ function make_PyTerminal(app: PyScriptApp) {
             return this.hasAttribute('auto');
         }
 
+        isDocked() {
+            return this.hasAttribute('docked');
+        }
+
         // implementation of the Stdio interface
         stdout_writeline(msg: string) {
             this.outElem.innerText += msg + '\n';
+            if (this.isDocked()) {
+                this.scrollTop = this.scrollHeight;
+            }
```

Comment:
dumb question: what are `scrollTop` and `scrollHeight`?

Reply:
[scrollTop](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTop)
[scrollHeight](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollHeight)

the TL;DR is that this forces the terminal to be scrolled to the bottom when docked and it doesn't gorw automatically, otherwise you need to ctrl+shift then scroll manually every single time ... I've thought this might be a better way, same way when I do stuff on the terminal I always end up to the last line.

does this make sense?

Reply:
That's a cool trick! 😄